### PR TITLE
Add Nameserver15 implementation

### PR DESCRIPTION
**Pull request by tgreenx**
## Purpose

This PR proposes an implementation of the newly specified [Nameserver15](https://github.com/zonemaster/zonemaster/blob/develop/docs/public/specifications/tests/Nameserver-TP/nameserver15.md) Test Case.

## Context

https://github.com/zonemaster/zonemaster/pull/1150

## Changes

`lib/Zonemaster/Engine/Test/Nameserver.pm` + `profile.json`
- Add implementation
- Add documentation

`docs/logentry_args.md`
- Update message arguments list

`t/Test-nameserver15.t` + `t/Test-nameserver15.data`
- Add unitary tests

## How to test this PR

Unit tests should pass. All message tags are covered, however note that they are not using "controlled" zones.

Also:
```
$ zonemaster-cli --show-testcase --test nameserver/nameserver15 zonemaster.net --level INFO
Seconds Level     Testcase       Message
======= ========= ============== =======
   0.00 INFO      UNSPECIFIED    Using version v4.6.2 of the Zonemaster engine.
   1.33 INFO      NAMESERVER15   The following name server(s) respond to software version query "version.bind" with string "9.18.13". Returned from name servers: "ns2.nic.fr/192.93.0.4;ns2.nic.fr/2001:660:3005:1::1:2"
   1.34 INFO      NAMESERVER15   The following name server(s) do not respond to software version queries. Returned from name servers: "nsa.dnsnode.net/194.58.192.46;nsa.dnsnode.net/2a01:3f1:46::53;nsp.dnsnode.net/194.58.198.32;nsp.dnsnode.net/2a01:3f1:3032::53;nsu.dnsnode.net/185.42.137.98;nsu.dnsnode.net/2a01:3f0:400::32"
```

### CODE REVIEW

**Review by blacksponge — APPROVED** [review]
It works as excepted from what I can see.

File: docs/logentry_args.md [hunk 1]
```diff
@@ -68,10 +68,12 @@ and updated messages (*msgids* and *msgstr*).
...
 | nsname         | Domain name                              | The domain name of a name server.                                       |
 | nsname_list    | List of domain names                     | A list of name servers, as specified by "nsname", separated by ";".     |
+| query_name     | Text                                     | A query domain name (QNAME), as defined in [RFC1035, section 4.1.2].    |
 | rcode          | An RCODE Name                            | An RCODE Name (not numeric code) from [DNS RCODEs].                     |
 | rrtype         | A Resource Record TYPE Name              | A Resource Record TYPE Name (not numeric code) from [DNS RR TYPEs].     |
 | soaserial      | Non-negative integer                     | The numeric value for the SERIAL field in an SOA record. Integer in range 0-4,294,967,295 |
 | soaserial_list | List of non-negative integers            | A list of non-negative integers, as specified by "soaserial", separated by ";". |
+| string         | Text                                     | The content of the RDATA of a TXT ressource record.                     |
```
Comment (matsduf) [inline docs/logentry_args.md:76]: "resource" is misspelled.

File: docs/logentry_args.md [hunk 2]
```diff
@@ -149,5 +151,6 @@ Message names maked with a question mark should not be considered stable.
 [DNSSEC algorithm]:                          https://www.iana.org/assignments/dns-sec-alg-numbers/dns-sec-alg-numbers.xhtml
 [DS Digest algorithm]:                       https://www.iana.org/assignments/ds-rr-types/ds-rr-types.xhtml
 [fr.po]:                                     ../share/fr.po
+[RFC1035, section 4.1.2]:                    https://datatracker.ietf.org/doc/html/rfc1035#section-4.1.2
 [sv.po]:                                     ../share/fr.po
```
Comment (matsduf) [inline docs/logentry_args.md:155]: Could you fix the typo here (not from this PR): `../share/fr.po` -->  `../share/sv.po`

**Review by matsduf — CHANGES_REQUESTED** [review]
Some minor comments, but in general it looks fine, but I have to test too.

**Review by blacksponge — APPROVED** [review]

**Review by matsduf — APPROVED** [review]

**Pull request merged**